### ISSUE TITLE
Account names changed unexpectedly

### ISSUE THREAD

**Issue by ilanolkies** [poanetwork/nifty-wallet#423]
### What browser and version did you use?

Chrome latest

### What operating system did you use?

Mac OS

### Do you have screenshots showing the problem?

![Screen Shot 2020-11-12 at 19 24 55](https://user-images.githubusercontent.com/36084092/99004293-96402e80-251d-11eb-9012-704ef720e2c6.png)

### What is your issue?

I started my wallet from scratch, with new sed, and created a new account (the second beside the default one that is created). I switched between networks and then went back to Ethereum. I found numbers 3 and 4 instead of 1 and 2. Then I create a third account and got number 3 again...

Expected behaviour:
- Accounts should never change nickname over time
- Accounts of different networks should have always different names, thus they map to different addresses

**Comment by vbaranov** [comment]
Duplicate of https://github.com/poanetwork/nifty-wallet/issues/265

**Comment by ilanolkies** [comment]
I dis not remove any account. This is a different case 🙂 maybe it helps reproducing it

**Comment by vbaranov** [comment]
@ilanolkies yes, thanks for the detailed description.

**Comment by manuelfr51** [comment]
Same here, and account 1 is missing, so balances not shown, restoring from seed fixes it for some time then again, and again

**Comment by ilanolkies** [comment]
@manuelfr51 I went to MyCrpyto desktop with my seed to recover and operate my funds meanwhile

**Comment by marianogappa** [comment]
This is also happening for me (edited account names are lost after changing networks), but two bigger issues are that:
- Upon changing networks and changing back, ERC20 addresses change!
- Upon restoring from seed phrase, only one account out of all created accounts is retrieved!

This is a big issue for me because my wallets are not empty!

**Comment by vbaranov** [comment]
>Upon restoring from seed phrase, only one account out of all created accounts is retrieved!

@marianogappa indeed, after restoring from seed phrase only first account appears in the dropdown. If you want to see other accounts derived from this seed phrase you can click `Create accounts` in the accounts menu.

>Upon changing networks and changing back, ERC20 addresses change!

What do you mean? Please elaborate in a separate issue. Let's keep this one for changing of account names.

**Comment by marianogappa** [comment]
@vbaranov Oh, ok. I've just done this and recovered the account I was missing. Thank you!

Personally, I think this is unintuitive behaviour: if you create a Nifty Wallet login and create say 5 accounts within it, if you then restore it somewhere else wouldn't you expect to see those 5 accounts? "Create account" is a confusing action, because  conceptually you're still "restoring" what you originally had. I suspect I won't be the last user coming here with this confusion.

Regarding the second issue, it's been addressed on a separate issue already by others, so I won't duplicate it. Cheers!
